docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc

```diff
@@ -1,33 +1,374 @@
-    print ("Single measurement value: [",measurementValue,"]")
+# Get last value of a Data Packet
+packet.last_value()  # TODO ()???
 ----
 ====
 
+[#printing_the_last_value]
+== Printing the last value
+
+No matter the source (either the Signal or the Data Packet), the resulting variable can be, for example, printed into the console.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Print last value
+std::cout << "lastValue: " << lastValue << std::endl;
+----
+Python::
++
+[source,python]
+----
+# Print last value
+print("last value: ", last_value)
+----
+====
+
+[#casting_the_last_value]
+== Casting the last value and using the cast variable
+
+In C++, if we would like to do something useful with the `IBaseObject`, such as access its underlying methods, we need to cast it into the correct type first. This is not necessary in Python, however.
+
+The underlying type of the `IBaseObject` we get from our `getLastValue` call can be cast into either `IFloat` if the type is `Float32` or `Float64`, to `IInteger` if the type is `Int8` through `Int64` or `UInt8` through `UInt64`, to `IComplexNumber` if type is `ComplexFloat32` or `ComplexFloat64`, to `IRange` if the type is `RangeInt64`, to `IStruct` if the type is `Struct`, and to `IList` of the aforementioned types if there is exactly one dimension in the single value.
+
+In this example, we cast `lastValue` object into an `IFloat` like so:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IFloat
+auto number = lastValue.asPtr<IFloat>();
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which we call `getLastValue` (`last_value` in Python) is:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// The Data Descriptor for Float64
+auto descriptor = DataDescriptorBuilder().setSampleType(SampleType::Float64).build();
+----
+Python::
++
+[source,python]
+----
+# The Data Descriptor for Float64
+builder = opendaq.DataDescriptorBuilder()
+builder.sample_type = opendaq.SampleType.Float64
+descriptor = builder.build()
+----
+====
+
+[#last_value_range]
+== Getting the last value with a Range
+
+In the following example, we cast the `lastValue` object into an `IRange` like so:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IRange
+auto range = lastValue.asPtr<IRange>();
+----
+====
+
+An example of something we could now do is call the methods of the underlying type and access its members such as the `getLowValue` (`low_value` in Python) and the `getHighValue` (`high_value` in Python), in this case provided by the `IRange` interface.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Extract values
+auto low = range.getLowValue();
+auto high = range.getHighValue();
+----
+Python::
++
+[source,python]
+----
+# Extract values
+low = last_value.low_value
+high = last_value.high_value
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which we call `getLastValue` (`last_value` in Python) is:
+
+[tabs]
```

Comment:
i would remove this example and add comment that we are assuming that type of data descriptor is int64